File: algorithms/K/Kids With the Greatest Number of Candies/Kids With the Greatest Number of Candies.js
var kidsWithCandies = function(candies, extraCandies) {
    //First find out maximum number in array:->
    let max=0, res=[]; 
    for(let i=0; i<candies.length; i++){
        if(candies[i]>max) max=candies[i];
    }
    console.log(max)
    
    /*Now add extraCandies with every element in array and checks if
	that sum is equals to or greater than max and return true and false otherwise; */
    
    for(let i=0; i<candies.length; i++){
        let temp=candies[i]+extraCandies;
        if(temp>=max) res.push(true);
        else res.push(false);
    }
    return res;
};

File: scripts/algorithms/K/Kids With the Greatest Number of Candies/Kids With the Greatest Number of Candies.py
class Solution:
    def kidsWithCandies(self, candy, extra):
        #create an array(res) with all values as True and it's lenght is same as candies
        res = [True]*len(candy)
        #iterate over the elements in the array candy
        for i in range(len(candy)):
            #if the no. of canides at curr position + extra is greater than or equal to the maximum of candies then continue 
            if (candy[i] + extra) >= max(candy):
                continue
            #if not 
            else:
                #change the value of that position in res as false
                res[i] = False
        #return the res list
        return res   